microsetta_admin/server.py

```diff
@@ -556,6 +574,126 @@ def new_kits():
                          mimetype='text/csv')
 
 
+@app.route('/add_barcode_to_kit', methods=['GET', 'POST'])
+def new_barcode_kit():
+
+    if request.method == 'GET':
+        return render_template('add_barcode_to_kit.html',
+                               **build_login_variables())
+
+    elif request.method == 'POST':
+        kit_ids = []
+        barcodes = []
+
+        if 'kit_ids' in request.form:
+            kit_ids = request.form['kit_ids']
+
+        if 'user_barcode' in request.form:
+            user_barcode = request.form['user_barcode']
+            barcodes.append(user_barcode)
+
+        if 'generate_barcode_single' in request.form:
+            generate_barcode_single = request.form['generate_barcode_single']
+        else:
+            generate_barcode_single = None
+
+        if 'generate_barcodes_multiple' in request.form:
+            generate_barcodes_multiple = request.form
+            ['generate_barcodes_multiple']
+        else:
+            generate_barcodes_multiple = None
+
+        if 'kit_ids' in request.files:
+            kit_ids_file = request.files['kit_ids']
+            kit_ids_content = kit_ids_file.read().decode('utf-8-sig')
+            kit_ids = [row[0] for row in csv.reader(io.StringIO
```

Comment:
This breaks if the CSV file has blank lines. It would be good to handle that more gracefully since exporting Excel files can sometimes lead to erroneous newlines.